src/plopp/backends/matplotlib/canvas.py

```diff
@@ -19,15 +19,13 @@ def _to_floats(x: np.ndarray) -> np.ndarray:
     return mdates.date2num(x) if np.issubdtype(x.dtype, np.datetime64) else x
 
 
-def _none_if_not_finite(x: Union[float, int, None]) -> Union[float, int, None]:
+def _none_if_not_finite(x: float | None) -> float | int | None:
```

Comment:
Why was `int` removed here (and other places)?

Reply:
Because `int | float` is the same as `float` to mypy. This is technically not clean because `int` is not a subclass of `float`. But mypy treats it as such. And ruff pointed that out.

Reply:
> But mypy treats it as such

I don't completely understand why we want to make mypy happy when it's asking for something which is not technically correct. Also, it seems weird that the `int` was removed from the input type, but not from the return type?

Reply:
> I don't completely understand why we want to make mypy happy when it's asking for something which is not technically correct. 

We have to make one of the type checkers happy, otherwise we aren't really testing out the type annotations. The reasoning is given in this PEP https://peps.python.org/pep-0484/#the-numeric-tower. We either open issues upstream or live with the fact `float` includes `int` with python type hints 🤷

Reply:
Ok for the tower, but it still doesn't answer why the return type still has `float | int`?

Reply:
I don't know. I can't think of a case where it would make a difference.
Technically, PEP484 only specifies this behaviour for arguments, not return types. But I tried it out with mypy and it seems to behave the same for the return type.